modules/new_user/new_user.py

```diff
@@ -24,24 +24,31 @@ async def add_role(ctx, machon, year):
 	print(f"Gave {new_role} to {member}")
     
 
-async def remove_unassigned(member):
+async def switch_assigned(member):
 	'''removes the unassigned role from member'''
-	role = get(member.guild.roles, id = int(get_id("UNASSIGNED_ROLE_ID")))
+	#remove unassigned role
+	role = get(member.guild.roles, id=get_id("UNASSIGNED_ROLE_ID"))
 	await member.remove_roles(role)
+
+	#add assigned role
```

Comment:
Add the assigned role before removing unassigned. (In case something goes wrong they'll still be able to use the welcome chat)